runtime/drivers/mysql/mysql.go

```diff
@@ -70,8 +76,24 @@ func (d driver) Open(instanceID string, config map[string]any, st *storage.Clien
 	if instanceID == "" {
 		return nil, errors.New("mysql driver can't be shared")
 	}
-	// actual db connection is opened during query
+
+	conf := &ConfigProperties{}
+	if err := mapstructure.WeakDecode(config, conf); err != nil {
+		return nil, fmt.Errorf("failed to decode config: %w", err)
+	}
+
+	if conf.DSN == "" {
+		return nil, fmt.Errorf("dsn not provided")
+	}
+
+	// Open DB handle
+	db, err := sqlx.Open("mysql", conf.DSN)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open connection: %w", err)
+	}
```

Comment:
Same comment as for postgres – avoid a permanently open handle since it's just for pings.